### ISSUE TITLE
Is the Capsule Protocol a premature abstraction?

### ISSUE THREAD

**Issue by mnot** [ietf-wg-masque/draft-ietf-masque-h3-datagram#159]
This protocol is being introduced with one use case, and that case is a generalisation that's capable of incorporating almost anything (datagrams). 

Is this a case of premature abstraction?

While from an implementation standpoint this abstraction might not be expensive, adding a new abstraction to HTTP's core data model isn't something we should do lightly; the road is littered with failed and poorly-integrated attempts.

**Comment by DavidSchinazi** [comment]
At the MASQUE Interim in April 2021, the WG reached consensus on strongly associating HTTP Datagrams with requests, and the fact that HTTP Datagrams would work across intermediaries. We also established that there was a need to register context semantics end-to-end.  That same day, Ben Schwartz proposed the name capsule for such a system and we've kept it since. In the past year, we've thoroughly discussed capsules among the Working Group, and the possibility of premature optimization was discussed. The Working Group decided that this was a useful abstraction, and CONNECT-IP and WebTransport both started using the abstraction, confirming that it was useful. The final design of the Capsule Protocol came out of the HTTP Datagrams design team, was presented at the February 2022 MASQUE Interim and Working Group consensus was declared by the chairs [on the mailing list last month](https://mailarchive.ietf.org/arch/msg/masque/GXaxmCJoo122ymqyAqTUaUFix_k/).

Unless you have any new information that would warrant revisiting this consensus, I'd suggest closing this issue with no action.
